src/library_fs.js

```diff
@@ -421,15 +421,45 @@ mergeInto(LibraryManager.library, {
           },
           isAppend: {
             get: function() { return (this.flags & {{{ cDefine('O_APPEND') }}}); }
-          }
+          },
+          flags: {
+            get: function() { return this.shared.flags; },
+            set: function(value) { this.shared.flags = value; }
+          },
+          position: {
+            get: function() { return this.shared.position; },
+            set: function(value) { this.shared.position = value; }
+          },
+          refcount: {
+            get: function() { return this.shared.refcount; },
+            set: function(value) { this.shared.refcount = value; }
+	  }
         });
       }
-      // clone it, so we can return an instance of FSStream
+
       var newStream = new FS.FSStream();
       for (var p in stream) {
-        newStream[p] = stream[p];
+        if (sharedProperties.indexOf(p) >= 0) {
+          if (!("shared" in newStream))
+            newStream.shared = {};
+          newStream.shared[p] = stream[p];
+        } else {
+          newStream[p] = stream[p];
+        }
+      }
+
+      if ("refcount" in newStream.shared) {
```

Comment:
From your explanation, I believe this is for NODERAWFS? If so, it can be behind `#if NODERAWFS`